### PR TITLE
Fixed wrong or missing PHPDoc

### PR DESCRIPTION
**Pull request by core23**
### Changelog

```markdown
### Fixed
- Fixed wrong or missing PHPDoc
```

### Subject
Some PHPDoc annotations are wrong or missing.

### CODE REVIEW

File: lib/Exporter/Source/AbstractXmlSourceIterator.php [hunk 1]
```diff
@@ -18,17 +18,59 @@
...
+     */
     protected $filename = null;
+
+    /**
+     * @var resource
+     */
     protected $file = null;
+
+    /**
+     * @var bool|null
```
Comment (OskarStark) [inline lib/Exporter/Source/AbstractXmlSourceIterator.php]: if we use `|null` here we need to add this to the other properties too

Comment (Soullivaneuh) [inline lib/Exporter/Source/AbstractXmlSourceIterator.php]: Why? Some properties should not have `null` value.

**Comment by OskarStark** [comment]
LGTM 👍

File: lib/Exporter/Source/ArraySourceIterator.php [hunk 2]
```diff
@@ -13,8 +13,6 @@
 
 class ArraySourceIterator extends IteratorSourceIterator
 {
-    protected $iterator;
```
Comment (Soullivaneuh) [inline lib/Exporter/Source/ArraySourceIterator.php]: This is BC break.

And why?

File: lib/Exporter/Source/AbstractXmlSourceIterator.php (hunk 1 above)
Comment (OskarStark) [inline lib/Exporter/Source/AbstractXmlSourceIterator.php]: check the property above

File: lib/Exporter/Writer/JsonWriter.php [hunk 3]
```diff
@@ -13,10 +13,19 @@
 
 class JsonWriter implements WriterInterface
 {
+    /**
+     * @var string
+     */
     protected $filename;
 
+    /**
+     * @var
```
Comment (Soullivaneuh) [inline lib/Exporter/Writer/JsonWriter.php]: This is the mysterious `@var`? :trollface:

File: lib/Exporter/Source/ArraySourceIterator.php (hunk 2 above)
Comment (core23) [inline lib/Exporter/Source/ArraySourceIterator.php]: The same property exists in `IteratorSourceIterator`, so no BC break .

File: lib/Exporter/Writer/JsonWriter.php (hunk 3 above)
Comment (core23) [inline lib/Exporter/Writer/JsonWriter.php]: It's an invisibly type ;)

File: lib/Exporter/Source/AbstractXmlSourceIterator.php (hunk 1 above)
Comment (core23) [inline lib/Exporter/Source/AbstractXmlSourceIterator.php]: Should I remove the `null` anotations or will we do this with another PR?

Comment (OskarStark) [inline lib/Exporter/Source/AbstractXmlSourceIterator.php]: i would add the null property in this PR

**Comment by Soullivaneuh** [comment]
:+1:
